### ISSUE TITLE
Add integration with `aws-vault` for credentials

### ISSUE THREAD

**Issue by kleinschmidt** [JuliaCloud/AWS.jl#612]
[`aws-vault`](https://github.com/99designs/aws-vault/) is convenient when using MFA, since it allows MFA-authenticated credentials to be cached for some time period, prompting the user for a new MFA when the credentials expire.  It's not immediately obvious to me whether the environment variable or server modes would be easier to integrate with how AWS.jl handles credentials; it looks like the ENV mode provides equivalent credentials to what AwsCredentials requires via environment variables, but AFAIU does not prompt for a refresh when the credentials expire.

**Comment by omus** [comment]
So using `aws-vault exec $role` directly will assume a role and set ENV variables that AWS.jl can use (as you mentioned). The problem comes in that these session variables will eventually expire resulting in you having to call `aws-vault exec $role` again to refresh the credentials which is extra annoying when working within the Julia REPL.

A better approach would be to use the AWS CLI supported [`credential_process`](https://github.com/99designs/aws-vault/blob/master/USAGE.md#use-case-3-aws-vault-is-a-mfa-session-cache-for-aws-sdk) with `aws-vault export` as shown in this [use case](https://github.com/99designs/aws-vault/blob/master/USAGE.md#use-case-3-aws-vault-is-a-mfa-session-cache-for-aws-sdk).

Support for `credential_process` was added in #620 (released in 1.85.0) which makes this approach feasible. 

My personal recommended way to set this up is to make you AWS CLI config (e.g. `~/.aws/config`) follow this pattern:

```
[profile JSMITH]
mfa_serial = arn:aws:iam::111222333444:mfa/JSMITH
credential_process = sh -c "unset AWS_VAULT; aws-vault export --format=json --prompt=terminal JSMITH"

[profile ROLE]
role_arn = arn:aws:iam::111222333444:role/ROLE
source_profile = JSMITH
include_profile = JSMITH

[profile CHAINED_ROLE]
role_arn = arn:aws:iam::111222333444:role/CHAINED_ROLE
source_profile = ROLE
```

Implementation details of the AWS CLI config file:
- We unset `AWS_VAULT` in [`credential_process`](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html) to allow the profile to be changed even when running within a `aws-vault exec` subshell.
- [`include_profile`](https://github.com/99designs/aws-vault/blob/master/USAGE.md#include_profile) is only recognized by `aws-vault` and is needed to inherit the `mfa_serial` setting.

Be sure to replace `JSMITH` with your AWS username and `ROLE` and `CHAINED_ROLE` with appropriate role names (if applicable). Be sure to remove any AWS CLI credential file (e.g. `~/.aws/credentials`) then the `credential_process` setting will be ignored.

If you want to fallback to a default profile it's recommended you use `AWS_PROFILE` for that purpose. Add the following to your user `rc` file (e.g. `~/.zshrc` for ZSH):

```sh
# Set the default profile. Avoid setting the profile when `aws-vault` is used as it interferes with the server mode
if [ -z "$AWS_VAULT" ]; then
    export AWS_PROFILE=JSMITH
fi

# OPTIONAL: Set some defaults for all profiles. The AWS CLI profile does not inherit any config 
# settings from the `source_profile` or the `[default]` profile.
# https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html#envvars-list
export AWS_DEFAULT_REGION="us-east-2"
export AWS_DEFAULT_OUTPUT="yaml-stream"  # Available formats: https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-output-format.html
export AWS_PAGER=""  # Disable use of a pager
```

Using this approach your MFA token will cached in your vault backend and the credentials can be shared across multiple Julia and AWS CLI sessions.

Finally, one issue I noticed is that using `AWSConfig(; profile)` doesn't actually set the override the profile when you run Julia from within an `aws-vault exec` session. This will be fixed in #621.

**Comment by omus** [comment]
TL;DR: Use AWS.jl 1.87.0 with the AWS configuration setting `credential_process = aws-vault export --format=json --prompt=terminal $user` for support for `aws-vault`.

As of AWS.jl 1.85.0 there is support for the AWS configuration [`credential_process` setting](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sourcing-external.html) which allows you to use [`aws-vault export` for handling credentials](https://github.com/99designs/aws-vault/blob/master/USAGE.md#use-case-3-aws-vault-is-a-mfa-session-cache-for-aws-sdk). Using this approach works well with AWS.jl as MFA tokens are cached on the `aws-vault` side and can be shared between the AWS CLI and AWS.jl.

I do recommend using AWS.jl 1.86.0 at a minimum though as this fixes an issue where `aws-vault exec` credentials were preferred over explicitly setting a profile via `global_aws_config(; profile)`. If you want to use `aws-vault exec --ecs-server` then use at least AWS.jl 1.87.0.
